src/ripple/core/impl/Config.cpp

```diff
@@ -212,12 +216,12 @@ void Config::setup (std::string const& strConf, bool bQuiet)
 
     QUIET       = bQuiet;
 
-    strDbPath           = Helpers::getDatabaseDirName ();
-    strConfFile         = strConf.empty () ? Helpers::getConfigFileName () : strConf;
-
-    VALIDATORS_BASE     = Helpers::getValidatorsFileName ();
+    strDbPath = databaseDirName;
 
-    VALIDATORS_URI      = boost::str (boost::format ("/%s") % VALIDATORS_BASE);
+    if (!strConf.empty())
+        strConfFile = strConf;
+    else
+        strConfFile = configFileName;
```

Comment:
I don't believe it is necessary to create a std::string out of a string literal to assign it. You should be able to assign the string literal.

strConfFile = strConf.empty() ? configFileName : strConf;